go-staticcheck.toml

```
@@ -105,6 +115,14 @@ rewrite='for range'
 match='for :[~_] = range'
 rewrite='for range'
 
+[S1005_03]
+match='for :[1], :[~_] = range'
+rewrite='for :[1] = range'
+
+[S1005_04]
+match='for :[1], :[~_] := range'
+rewrite='for :[1] := range'
+
```

```suggestion
[S1005_03]
match='for :[1:e], :[~_] = range'
rewrite='for :[1] = range'

[S1005_04]
match='for :[1:e], :[~_] := range'
rewrite='for :[1] := range'

```

Actually,  looking at this a bit more carefully I think there's a risk of matching across newlines and suggesting something incorrect. This will safely match just one expression.

---

thank you. updated